bin/clover/src/commands/inferAi.ts

```diff
@@ -0,0 +1,186 @@
+import { allCfProps, CfProperty, CfSchema, loadCfDatabase } from "../cfDb.ts";
+import OpenAI from "openai";
+import _ from "lodash";
+import { Inferred, loadInferred, saveInferred } from "../spec/inferred.ts";
+
+const SAVE_INTERVAL = 5;
+const GENERIC_PROMPT = `
+You are a cloud engineer working with AWS CloudFormation. Given the CloudFormation documentation
+for a resource property, you will decide whether the property has a known (limited) set of
+possible values. If it does, you will respond with a list of those values in a JSON array of strings.
+If it does not, you will respond with a JSON null value.
+
+These values will be used to drive a UI to edit this property: if you respond with a JSON array,
+the user will be presented with a dropdown list from which they can select only those values.
+It would be very bad if the field actually allowed other values than the ones you specify,
+because they would be unable to enter those values. So don't respond with an array unless you
+are absolutely sure.
+
+For example, given this CloudFormation documentation for the "Mode" property of AWS::ApiGateway::RestApi:
+
+  This property applies only when you use OpenAPI to define your REST API. The \`\`Mode\`\` determines how API Gateway handles resource updates.
+  Valid values are \`\`overwrite\`\` or \`\`merge\`\`.
+  For \`\`overwrite\`\`, the new API definition replaces the existing one. The existing API identifier remains unchanged.
+  For \`\`merge\`\`, the new API definition is merged with the existing API.
+  If you don't specify this property, a default value is chosen. For REST APIs created before March 29, 2021, the default is \`\`overwrite\`\`. For REST APIs created after March 29, 2021, the new API definition takes precedence, but any container types such as endpoint configurations and binary media types are merged with the existing API.
+  Use the default mode to define top-level \`\`RestApi\`\` properties in addition to using OpenAPI. Generally, it's preferred to use API Gateway's OpenAPI extensions to model these properties.
+
+You would respond with:
+
+  [ "overwrite", "merge" ]
+
+You must respond *only* with a JSON array of strings, or a JSON null value, and no other text.
+Do not include formatting or markdown delimiters. Do not include invalid JSON, and triple check
```

Comment:
Good shoutout

Reply:
That is actually how we fixed a bug :) "Stop doing that bug."